plugins/modules/configuration_item_batch.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: configuration_item_batch
+
+author:
+  - Manca Bizjak (@mancabizjak)
+  - Miha Dolinar (@mdolin)
+  - Tadej Borovsak (@tadeboro)
+
+short_description: Manage ServiceNow configuration items
+
+description:
+  - Create, delete or update a ServiceNow configuration item.
+  - For more information, refer to the ServiceNow configuration management documentation at
+    U(https://docs.servicenow.com/bundle/paris-servicenow-platform/page/product/configuration-management/reference/cmdb-table-property-descriptions.html).
+version_added: 1.0.0
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+
+seealso:
+  - module: servicenow.itsm.configuration_item_info
+
+options:
+  sys_class_name:
+    description:
+      - The sys_class_name parameter value contains a table name (configuration item type)
+        that we would like to manipulate. In our example, we are manipulating items in the
+        cmdb_ci_ec2_instance table.
+    required: true
+    type: str
+  id_column:
+    description:
+      - The id_column parameter contains a configuration item column name that should be
+        used to determine if we need to update an existing configuration item or if we must
+        create a new one. In the example above, we are using instance ID as a key, but we could
+        also use things like VM names (if they are unique), for example.
+    required: true
+    type: str
+  dataset:
+    description:
+      - The dataset parameter holds a list of dictionaries that contain source data.
+        Each item in the dataset list contains information about a configuration item we would
+        like to create or update. In our case, we passed it array of instances that we retrieved
+        from the AWS using info module.
+    required: true
+    type: list
+    elements: dict
+  map:
+    description:
+      - The map parameter contains a dictionary that informs module how to map source values to
+        configuration item attributes. Keys represent the configuration item column names and the
+        values are Jinja expressions that return the approriate value.
+      - Casting can be used on objects in parameters for the correct type.
```

Comment:
I am having some second thoughts about what to return by default. Since ServiceNow expects string in almost all situations, maybe we should default to string and only cast things if the user specifies a filter at the end of the expression?

```suggestion
      - Transformation instructions on how to convert input data to CMDB items.
      - Keys represent the CMDB item column names and the values are Jinja expressions that extract the value from the source data.
      - To cast the value to the right type, use I(string), I(int), or I(float) Jinja filters at the end of the expression.
      - If there is no filter at the end of the expression, data is returned as ???????.
```